ffi/cpp/include/decimal.hpp

```diff
@@ -0,0 +1,78 @@
+// This file is part of ICU4X. For terms of use, please see the file
```

Comment:
Praise: I haven't seen `*.hpp` before but I like it a lot, especially here where we have both C and C++ header files.